api_docs: Update instructions to deploy multiple bots using Botserver.

Niloth-p opened the pull request:
- Simplified the `botserverrc` description, and removed 2 redundant code blocks.
- Added the `bot-config-file` option to the `botserverrc` section example, with a short description.
- Edited the previously vague instruction to download the `botserverrc`, by linking to the respective doc. Linked the "Download zuliprc" instruction in the "Running a single bot" section to the doc as well.
- Fixed the incomplete command to run Botserver.

Fixes: #32793

**Screenshots and screen captures:**
![img](https://github.com/user-attachments/assets/c249c2f1-d598-46bb-9d78-562eebbd7db1)

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
@laurynmm would be great to get your review on this one.

laurynmm commented on api_docs/deploying-bots.md:
```diff
@@ -117,55 +117,44 @@ Botserver process.  You can do this with the following procedure.
+1. Download the `botserverrc` from your Bot settings page, by following
+   the first 3 steps from [these instructions][zuliprc], and using the
+   "**Download config of all active outgoing webhook bots in Zulip Botserver
+   format**" option.
```
This option is only shown if the user has at least one "Outgoing" webhook bot configured, so that might be worth noting somewhere here (e.g. in the dev environment it's only shown for aaron).

laurynmm commented on api_docs/deploying-bots.md:
````diff
@@ -117,55 +117,44 @@ Botserver process.  You can do this with the following procedure.
-    site=http://hostname
-    ```
+    For each bot, enter the name of the bot you want to run in the square
+    brackets `[]`, e.g., the above section applies to the `helloworld` bot.
````
-> "the above example applies to ..."

laurynmm reviewed (commented):
@Niloth-p - Thanks for updating this documentation! Had two small notes about these changes, but overall they look good.

Since #32793 is a pull request and not an issue, it would make more sense to note that these changes build on that work vs "fix" it.

Let me know if you have any questions or when this is ready for me to take another look.

Niloth-p commented:
@laurynmm - Thanks for the fast review!

> Since #32793 is a pull request and not an issue, it would make more sense to note that these changes build on that work vs "fix" it.

I added that for GitHub to auto-close that PR upon merge of this PR.
Which is why I didn't add the "Fixes" part to my commit message, and added it only to the PR description.

I've now slightly updated the "Fixes" line in the PR description to clarify the connection, within parentheses.

> This option is only shown if the user has at least one "Outgoing" webhook bot configured, so that might be worth noting somewhere here.

True, I myself went looking for that option, and then verifying that it still exists in the source code.
But, I didn't want to keep on expanding this PR (since the whole doc seems fairly outdated), when all it needed was the single option addition, so I hadn't focused on adding that earlier.

I've now added the usual "Add bot" instruction, and instead removed the link to "Downloading zuliprc", which doesn't fully apply to this case, and wasn't looking clean with the "Follow x steps from here" where x differs on zulip.com and `main` branch.
And removed the link that I added to the `Running a bot using the Zulip Botserver` section as well, localizing the changes to a single section.

Updated the commit message and PR description.

Please take another look, thank you!
You should be able to easily view the diffs between updates since I haven't rebased the branch.

laurynmm commented:
> I added that for GitHub to auto-close that PR upon merge of this PR. Which is why I didn't add the "Fixes" part to my commit message, and added it only to the PR description.
> 
> I've now slightly updated the "Fixes" line in the PR description to clarify the connection, within parentheses.

Cool. I didn't know that would close a linked PR automatically. As a reviewer, I usually start by looking at any linked "Fixes" notes in the PR description to get an idea for the issue that's being addressed in the changes, so I was confused by seeing another pull request vs an issue.

---

When I was reading through this API doc page in general, it seemed like we could pull out the instructions for creating an outgoing webhook in the web-app to it's own block, and then reuse that for both the Botserver instruction blocks (running a single bot and multiple bots).

So, I've made those updates and pushed them here (see updated screenshots below). It'd be great if you could review my updates, and let me know if you think there are any other bits to improve in this pass for these sections of the documentaiton.

---

**Updated screenshots**:

[CURRENT DOCUMENTATION](https://zulip.com/api/deploying-bots)
<details>
<summary>Zulip Botserver documentation</summary>

![Screenshot from 2025-01-06 14-28-31](https://github.com/user-attachments/assets/2df1816e-48c9-4bf4-9c02-7f4f3516a01f)
![Screenshot from 2025-01-06 14-28-47](https://github.com/user-attachments/assets/bb2b88d8-5940-4c19-9e01-18f7e7061b61)
</details>

Niloth-p commented:
That's some great re-organization! Simplifies everything, thank you!

Made multiple changes. Attached the diff.
And the main changes are outlined section-wise in the commit message.

<details>
<summary><h3>Screenshots</h3></summary>

![image](https://github.com/user-attachments/assets/6e013adf-03b8-4de3-8376-eef05a5e7250)
![image](https://github.com/user-attachments/assets/ebe8fa4b-925f-4d40-a8b0-6de7d4a76f39)
![image](https://github.com/user-attachments/assets/5979749a-c2a1-481d-9708-c978cb146295)


</details>

<details>
<summary><h3>The diff</h3></summary>

```diff
diff --git a/api_docs/deploying-bots.md b/api_docs/deploying-bots.md
index 22299ce17b..445bced58e 100644
--- a/api_docs/deploying-bots.md
+++ b/api_docs/deploying-bots.md
@@ -42,7 +42,7 @@ Botserver interaction are:
     @**My Bot User** hello world
     ```
 
-1. The Zulip server sends a POST request to the Botserver on `https://bot-server.ex
ample.com/`:
+1. The Zulip server sends a POST request to your Botserver endpoint URL:
 
     ```
     {
@@ -60,9 +60,7 @@ Botserver interaction are:
 1. The Botserver searches for a bot to handle the message, and executes your
    bot's `handle_message` code.
 
-Your bot's code should work just like it does with `zulip-run-bot`;
-for example, you reply using
-[bot_handler.send_reply](writing-bots#bot_handlersend_reply)).
+Your bot's code should work just like it does with `zulip-run-bot`.
 
 ### Installing the Zulip Botserver
 
@@ -76,27 +74,17 @@ pip3 install zulip_botserver
 
 {start_tabs}
 
-1. Construct the URL for your bot, which will be of the form:
+1. Navigate to the [Bots tab][your-bots] of the **Personal settings** menu,
+   and click **Add a new bot**.
 
-    ```
-    http://<hostname>:<port>
-    ```
-
-    The `hostname` is the hostname you'll be running the Botserver on,
-    and `port` is the port for it. The recommended default port is
-    `5002`.
-
-1. Log in to your Zulip organization.
+1. Set the **Bot type** to **Outgoing webhook**.
 
-1. Via the gear menu, navigate to **Personal settings
-   (<i class="zulip-icon zulip-icon-gear"></i>)** -> **Bots**, and click
-   **Add a new bot**.
+1. Set the **endpoint URL** to `https://<host>:<port>` where `host` is the
+   hostname of the server you'll be running the Botserver on, and `port` is
+   the port number. The default port is `5002`.
 
-1. Select **Outgoing webhook** for bot type, and fill out the form using
-   the URL constructed above.
-
-1. Click **Create bot**. You should see the new bot user in the **Active
-   bots** panel.
+1. Click **Create bot**. You should see the new bot user in the
+   **Active bots** panel.
 
 {end_tabs}
 
@@ -104,12 +92,12 @@ pip3 install zulip_botserver
 
 {start_tabs}
 
-1. [Create your bot](#create-a-bot-in-your-zulip-organization)
-   in your Zulip organizaiton.
+1. [Create your bot](#create-a-bot-in-your-zulip-organization) in your Zulip

+   organization.
 
 1. Download the `zuliprc` file for the bot created above from the
-   **Active Bots** panel in the **Bots** tab of your **Personal Settings**,
-   using the download button.
+   [Bots tab][your-bots], by clicking the download
+   (<i class="fa fa-download"></i>) icon under the bot's name.
 
 1. Run the Botserver, where `helloworld` is the name of the bot you
    want to run:
@@ -134,9 +122,9 @@ Botserver process.
 1. [Create your bots](#create-a-bot-in-your-zulip-organization)
    in your Zulip organization.
 
-1. Download the `botserverrc` file from the **Bots** tab of your **Personal
-   Settings**, using the **Download config of all active outgoing webhook
-   bots in Zulip Botserver format** option.
+1. Download the `botserverrc` file from the [Bots tab][your-bots], using
+   the **Download config of all active outgoing webhook bots in Zulip
+   Botserver format** option.
 
 1. Open the `botserverrc`. It should contain one or more sections that look
    like this:
@@ -155,21 +143,25 @@ Botserver process.
 1. For each bot, enter the name of the bot you want to run in the square
    brackets `[]`, e.g., the above example applies to the `helloworld` bot.
    To run an external bot, enter the path to the bot's python file instead,
-   e.g., `[~/Documents/my_new_bot.py]`.
+   e.g., `[~/Documents/my_bot_script.py]`.
+
+    !!! tip ""
 
-    Note that the `bot-config-file` is needed only for bots that use a config
-    file.
+        The `bot-config-file` setting is needed only for bots that
+        use a config file.
 
 1.  Run the Zulip Botserver by passing the `botserverrc` to it.
 
      ```
-     zulip-botserver --config-file <path to botserverrc> --hostname <address> --por
t <port>
+     zulip-botserver --config-file <path-to-botserverrc> --hostname <address>
 --port <port>
      ```
 
      If omitted, `hostname` defaults to `127.0.0.1` and `port` to `5002`.
 
 {end_tabs}
 
+[your-bots]: {{zulip_url}}/#settings/your-bots
+
 ### Running Zulip Botserver with supervisord
 
 [supervisord](http://supervisord.org/) is a popular tool for running

```

</details>

laurynmm commented:
@Niloth-p - Sure, that makes sense. Marking this for integration review.

I think you could probably revise the commit message and simplify it a bit. I'm not sure it's super useful to have all the changes listed out there. A more general summary of the updates might be more useful for someone who's looking back through the commit history for this doc.

timabbott commented:
Looks great, merged, thanks @Niloth-p and @laurynmm!
